doc/manual/source/glossary.md

```diff
@@ -31,9 +31,24 @@
 
   The industry term for storage and retrieval systems using [content addressing](#gloss-content-address). A Nix store also has [input addressing](#gloss-input-addressed-store-object), and metadata.
 
+- [derivation]{#gloss-derivation}
+
+  A derivation can be thought of as a [pure function](https://en.wikipedia.org/wiki/Pure_function) that produces new files from existing files, taking into account how these files refer to each other.
+
+  In Nix, linked files are represented as [store objects][store object].
+  Derivations are implemented as [operating system processes that run in a sandbox](@docroot@/store/building.md#builder-execution).
```

Comment:
Maybe this can be phrased more concisely as something like

```suggestion
  A derivation is an operating system process that produces store objects from other store objects.
```

Reply:
That's essentially what I started with, but it lacks the distinctive characteristics of derivations. Otherwise readers will rightfully wonder why one would bother, since makefiles and build scripts are also describe processes that produce files (and may as well produce store objects if instructed).

Also, before jumping right into store objects we need a bit of context to pick up readers from what they already know: files. But again, the distinctive property of store objects is that they introduce references between these files. I'm trying to avoid the glossary term "reference" though, because that would be skipping ahead.